lampod/src/jsonrpc/channels.rs

```diff
@@ -8,3 +16,69 @@ pub fn json_list_channels(ctx: &LampoDeamon, request: &json::Value) -> Result<js
     let resp = ctx.channel_manager().list_channel();
     Ok(json::to_value(resp)?)
 }
+
+pub fn json_close_channel(ctx: &LampoDeamon, request: &json::Value) -> Result<json::Value, Error> {
+    log::info!("call for `closechannel` with request {:?}", request);
+    let mut request: request::CloseChannel = json::from_value(request.clone())?;
+    let events = ctx.handler().events();
+    // FIXME: There could be a better way to do this
+    let mut res = Ok(());
+    if request.channel_id.is_none() {
+        let channels: response::Channels = ctx.handler().call("channels", json::json!({})).unwrap();
+
+        let mut vis = false;
+        for channel in channels.channels.iter() {
+            if channel.peer_id == request.node_id {
+                vis = true;
+                request.channel_id = Some(channel.channel_id.clone());
+                res = ctx.channel_manager().close_channel(request.clone());
+                break;
+            }
+        }
+        // When we do not get any channel_id associated with node_id we throw an error
+        if !vis {
+            res = Err(Error::Rpc(RpcError {
+                code: -1,
+                message: format!("Not found"),
+                data: None,
+            })
+            .into());
+        }
+    } else {
+        res = ctx.channel_manager().close_channel(request.clone());
+    }
```

Comment:
```suggestion
        let channels: response::Channels = ctx.handler().call("channels", json::json!({
        "peer_id": request.node_id,
        }))?; // DO NOT UNWRAP
        if channels.len() > 1 {
           // check the channel_id if it is not none, if it is return an error
        } else if !channels.is_empty() {
          /// SAFETY: it is safe to unwrap because the channels is not empity
          let channel = channels.fist().unwrap();
           res = ctx.channel_manager().close_channel(channel.channel_id);
        }
```

This a bug free code that it is way batter than your complex code :)

Reply:
This would be the complete code I guess. Can you please confirm.
```rust
    if channels.channels.len() > 1 {
        // check the channel_id if it is not none, if it is return an error
        // and if it is not none then we need to have the channel_id that needs to be shut
        if request.channel_id.is_none() {
            return Err(Error::Rpc(RpcError {
                code: -1,
                message: format!("Provide channel_id"),
                data: None,
            }));
        } else {
            res = ctx.channel_manager().close_channel(request.clone());
        };
    } else if !channels.channels.is_empty() {
        // This is the case where channel with the given node_id = 1
        // SAFETY: it is safe to unwrap because the channels is not empty
        let channel = channels.channels.first().unwrap();
        request.channel_id = Some(channel.channel_id.clone());
        res = ctx.channel_manager().close_channel(request);
    } else {
        // No channels with the given peer.
        return Err(Error::Rpc(RpcError {
            code: -1,
            message: format!("No channels with associated peer"),
            data: None,
        }));
    };
```

Reply:
implement it and then we will see, there is no need of confirming anything